electrum/gui/kivy/uix/dialogs/request_dialog.py

```diff
@@ -159,25 +179,47 @@ def on_mode(self, instance, x):
             self.ids.qr.opacity = 1
         else:
             self.ids.qr.opacity = 0
-        self.update_status()
+        if not qr_data and self.error_text:
+            Clock.schedule_once(lambda dt: self.ids.qrdata_tabs.switch_to(self.ids.qrdata_tab_error))
+        else:
+            Clock.schedule_once(lambda dt: self.ids.qrdata_tabs.switch_to(self.ids.qrdata_tab_qr))
 
     def update_status(self):
         req = self.app.wallet.get_request(self.key)
+        help_texts = self.app.wallet.get_help_texts_for_receive_request(req)
+        address = req.get_address() or ''
+        URI = self.app.wallet.get_request_URI(req) or ''
+        lnaddr = req.lightning_invoice or ''
         self.status = self.app.wallet.get_request_status(self.key)
         self.status_str = req.get_status_str(self.status)
         self.status_color = pr_color[self.status]
         self.has_lightning = req.is_lightning()
-        warning = ''
-        if self.status == PR_UNPAID and self.mode == self.MODE_LIGHTNING and self.app.wallet.lnworker:
-            if self.amount_sat and self.amount_sat > self.app.wallet.lnworker.num_sats_can_receive():
-                warning = _('Warning') + ': ' + _('This amount exceeds the maximum you can currently receive with your channels')
-        if not self.mode == self.MODE_LIGHTNING:
-            address = req.get_address()
-            if not address:
-                warning = _('Warning') + ': ' + _('This request cannot be paid on-chain')
-            elif self.app.wallet.adb.is_used(address):
-                warning = _('Warning') + ': ' + _('This address is being reused')
-        self.warning = warning
+
+        self.warning = ''
+        self.error_text = ''
+        self.data = ''
+        if self.mode == self.MODE_ADDRESS:
+            if help_texts.address_is_error:
+                self.error_text = help_texts.address_help
+            else:
+                self.data = address
+                self.warning = help_texts.address_help
+        elif self.mode == self.MODE_URI:
+            if help_texts.URI_is_error:
+                self.error_text = help_texts.URI_help
+            else:
+                self.data = URI
+                self.warning = help_texts.URI_help
+        elif self.mode == self.MODE_LIGHTNING:
+            if help_texts.ln_is_error:
+                self.error_text = help_texts.ln_help
+            else:
+                self.data = lnaddr
+                self.warning = help_texts.ln_help
+        else:
+            raise Exception(f"unexpected {self.mode=!r}")
+        if self.warning:
+            self.warning = _('Warning') + ': ' + self.warning
```

Comment:
Note that self.warning is a kivy property. 
The GUI will be updated everytime it is changed.
This is why the original code was setting a local variable first.

Reply:
The same applies to `self.error_text`, btw

Reply:
oh right, indeed